src/psyclone/tests/psyGen_test.py

```diff
@@ -2247,3 +2247,8 @@ def test_siblings():
         siblings = loop.siblings
         assert loop in siblings
         assert len(siblings) == (1 if loop.ancestor(Loop) else 3)
+
+    # Special case of a root node
+    root_siblings = invoke.schedule.siblings
+    assert len(root_siblings) == 1
+    assert root_siblings[0] is invoke.schedule
```

Comment:
We typically require that the tests must be in the same directory path (but under /tests) than the file they are testing, so they are easier to find.

Tests for `src/psyclone/psyir/nodes/node.py`
must be in: `src/psyclone/tests/psyir/nodes/node_test.py`

Can you move test_sibiling there please?

Also, for testing core psyir functionality we try to avoid using any of the APIs (e.g. nemo) and avoid using external files, so instead of:
`_, invoke = get_invoke("explicit_do_two_loops.f90", "nemo", 0)`

You could use the fortran_reader fixture to create the specific psyir that you need for the test:
```
def test_siblings(fortran_reader):
    code = '''subroutine test()
        ! Code needed for this test
    end subroutine'''
    psyir = fortran_reader.psyir_from_source(code)
```
You can see an example in the last test on that file (`test_path_from`)

Reply:
Thanks, have now addressed this in https://github.com/stfc/PSyclone/pull/2380/commits/af68f179ba073e688619a8b9f4c7c5c59159d6c8.